File: ll.go
package main

import "fmt"

type Node struct {
	value int
	next  *Node
}

type LinkedList struct {
	head *Node
}

func (l *LinkedList) Append(value int) {
	node := &Node{value: value}

	if l.head == nil {
		l.head = node
		return
	}

	current := l.head

	for current.next != nil {
		current = current.next
	}

	current.next = node
}

func (l *LinkedList) Print() {
	current := l.head
	for current != nil {
		fmt.Println(current.value)
		current = current.next
	}
}

File: ll.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

class Linkedlist {
    constructor(){
        this.head = null
    }

    append(value) {
        const node = new Node(value);
        if (!this.head) {
            this.head = node;
            return;
        }
        let current = this.head;

        while (current.next) {
            current = current.next;
        }
        current.next = node;
    }

    print(){
        let current = this.head;
        while(current) {
            console.log(current.value);
            current = current.next;
        }
    }


}